### ISSUE TITLE
Street Name/Municipality Name in Reports & Notices

### ISSUE THREAD

**Issue by gk-tl** [bcgov/smk-tlink#52]
TL BETA used the street name/municipality from the DRA and tied them to the corresponding report/notice. TRP does not currently do this. As a result, the list of notices for a generated route doesn't provide sufficient context for the user.

**See example below:**

![image](https://user-images.githubusercontent.com/59627982/73485136-6a213b80-4357-11ea-8a9c-b878408426a0.png)

May also have an impact on #13

**Comment by gk-tl** [comment]
Please note - an alternate approach (and likely easier to implement) would be to include Street Name + Municipality info as part of the report/notice text. 

Example: 

**108th Avenue between 152 Street and King George Boulevard, Surrey:** OS-OW Vehicle Restriction – No oversize vehicles greater than 3.2 metres in width, inclusive of load.

Please let me know if this is preferred, in which case I will update the Restriction/Advisory/Application Messages accordingly.

@NicoledeGreef @qqnluaq

**Comment by NicoledeGreef** [comment]
I think it makes sense to include them in the text if there is no need to cross-reference.  What do you think @qqnluaq?

**Comment by cmhodgson** [comment]
It seems to me that this would make maintaining the text much more complicated. I have two approaches in mind:
1) add the notifications to the relevant direction (which includes the street name). This would allow the notifications to be included in the directions for maximum context, and they could also be summarized in a report section fairly easily. The directions do not presently include the municipality, we could change them so they do, but that might require introducing new directions at municipal boundary crossings.
2) add the street name and municipality to the notifications as additional properties

They are about the same minor level of difficulty, (1) is more powerful but would require a little work from Ben to build the report summary, (2) is pretty straightforward, but I like (1) because we are using the location info from the directions instead of written words to identify the location.

**Comment by mraross** [comment]
From a corporate route planner API perspective, I lean toward option 1 because it is more useful to any application that might use the API (e.g., OnRoute).

**Comment by gk-tl** [comment]
Thank you for reviewing this card so quickly. If approach 1 works for everyone, we support it (without the need to identify the municipality). Happy to make approach 2 work if need be.

**Comment by NicoledeGreef** [comment]
Sounds like the Route Planner will supply - no worries here.

**Comment by gk-tl** [comment]
To clarify - would the additional context (e.g., Street Name) also be provided in a summary report in the directions printout and an "on-screen" report as mentioned in #13? 

> In the TL BETA, reports are a stand-alone feature, but the TL BETA doesn't provide turn-by-turn directions...happy to receive a recommendation about incorporating reports in the full production TRP, and by extension, how to display reports on the route detail printout.

TL BETA Reports
![image](https://user-images.githubusercontent.com/59627982/73496599-cbeca000-436d-11ea-8530-05703eb0e895.png)

**Comment by cmhodgson** [comment]
In order to build the reports list, the UI code would just have to loop through the directions and pull out the relevant truck notifications, at which point it can also pull out the associated street name, and build the report content from that. Should be a trivial bit of work on the UI side to build this summary.

**Comment by NicoledeGreef** [comment]
@qqnluaq  -  all good? relative estimate of effort required please.

**Comment by cmhodgson** [comment]
Below is an example of the directions going up georgia st and across the lions gate bridge. Because the last bit of georgia st has the notification on it, the entire direction associated with georgia st (1.7 km of road in this case) has the notification attached to it. You also do get the same notification repeated multiple times for each differently named street, but I guess that was how it worked before, as well? Just want to make sure this is OK.

Directions:
Head northwest on Georgia St for 1.7 km (2 minutes 56 seconds)
 - Vehicles with a GVW greater than 13,000 kg are prohibited on the Lions Gate Bridge
Continue onto Stanley Park Causeway for 2.2 km (2 minutes 9 seconds)
 - Vehicles with a GVW greater than 13,000 kg are prohibited on the Lions Gate Bridge
Continue onto Lions Gate Bridge for 2 km (1 minute 59 seconds)
 - Vehicles with a GVW greater than 13,000 kg are prohibited on the Lions Gate Bridge
Continue onto Marine Dr Onramp for 300 m (28 seconds)
 - Vehicles with a GVW greater than 13,000 kg are prohibited on the Lions Gate Bridge
Continue onto Marine Dr for 200 m (13 seconds)
Continue onto Westbound Capilano Bridge for 150 m (10 seconds)
Continue onto Marine Dr for 25 m (2 seconds)
Finish!

**Comment by gk-tl** [comment]
Repeating the same notices/restrictions multiple times may become an issue as some of them can be lengthy. Based on the example above, I will also need to rewrite most of the notices to mirror the structure of the one for Lions Gate (at a minimum include the road name). 

eg. “OS-OW Vehicle Restriction – No oversize vehicles greater than 3.2 metres in width, inclusive of load allowed **on 108th Avenue**” instead of “OS-OW Vehicle Restriction – No oversize vehicles greater than 3.2 metres in width, inclusive of load allowed.”

In other words, basically option 2. Otherwise, there will be cases when the restrictions in the directions won’t make sense. Take out “Lions Gate Bridge” in the example above and stop at “Vehicles with a GVW over 13,000 kg are prohibited”. This would imply the weight restriction applies to all of Georgia, which is not the case. There will likely be similar issues for the other restrictions/notices. 

Happy to revise the text in the EXCEL file we previously provided, but pointing out that we would be implementing option 1 and 2.

**Comment by cmhodgson** [comment]
Such a notice would be attached to a direction which has the street name in it. It should be easy for the UI to grab the name from the direction and add it, as in "On <street name>, <Restriction text>". However for text that already includes a street name such as the lions gate above, that would be a little confusing and overkill - however, your previous system must have been doing this same approach, no? 
Perhaps changing the text is necessary for the best result, however, if you are going to be re-writing the notices, you should discuss this with Graeme as he will need to re-attach the notices to the segments - we would want to make sure we can do this efficiently.

**Comment by qqnluaq** [comment]
I'm more concerned about the distinction between notifications for the entire route, vs notifications for a particular direction. 

If it is important that the 2 types of notifications be displayed differently (as they are on the printout at present), then we should keep them separate. So I'd advocate for option #2 above, and attach the address to the route notification as appropriate.

Modifying the printout to handle the address is a small amount of work, say 4 hours.

How the notifications are to be displayed on the application UI is still an open question AFAIK, I don't see a ticket that addresses that issue.

**Comment by cmhodgson** [comment]
I just had a call with Ben, I think we may need to have a call next week with the Translink folks to figure out how best to handle this.

**Comment by gk-tl** [comment]
Happy to chat as it'll be probably be more efficient to work this out on a call. Monday, Feb 3, 3-4 pm or Tuesday, 2-3 pm would work on our end.

**Comment by gk-tl** [comment]
In response to Ben's comment - it's fairly important they be kept separate. Some of the notices apply to the entire route (e.g., you may require a permit if vehicle parameters exceed legislated limits) and some are specific to route segments (no OS-OW vehicle/loads on the Pattullo Bridge). 

I don't think there is a distinct ticket re. addressing notifications in the UI, and while it has come up in the comments for other tickets (e.g., #13), it is very much an open question. Please correct me if I'm wrong, but the approach we took in the TL BETA (a modal page with Reports & Notices) should be fairly straight forward to implement.

> I'm more concerned about the distinction between notifications for the entire route, vs notifications for a particular direction.
> 
> If it is important that the 2 types of notifications be displayed differently (as they are on the printout at present), then we should keep them separate. So I'd advocate for option #2 above, and attach the address to the route notification as appropriate.
> 
> Modifying the printout to handle the address is a small amount of work, say 4 hours.
> 
> How the notifications are to be displayed on the application UI is still an open question AFAIK, I don't see a ticket that addresses that issue.

**Comment by gk-tl** [comment]
Will it be possible to produce a summary list like this when we include the segment specific notices in line with the directions, and make this list available on the last page of the directions and a modal in the app?

![image](https://user-images.githubusercontent.com/59627982/73886694-f1baee80-481e-11ea-9b70-70291457c03c.png)

**Comment by NicoledeGreef** [comment]
20200205 call: Include street name (not muni) in directions output from route planner, and ultimately include in the printable and modal window reports.

Include Restrictions/advisories/notifications icon in the list of route directions and the printable directions.

**Comment by gk-tl** [comment]
Complete. Thank you.

**Comment by gk-tl** [comment]
Complete. Thank you.
